src/Controller/HelperController.php

```diff
@@ -285,32 +285,26 @@ public function setObjectFieldValueAction(Request $request)
 
         // Handle entity choice association type, transforming the value into entity
         if ('' !== $value && 'choice' == $fieldDescription->getType() && $fieldDescription->getOption('class')) {
-            // Get existing associations for current object
-            $associations = $admin->getModelManager()
-                ->getEntityManager($admin->getClass())->getClassMetadata($admin->getClass())
-                ->getAssociationNames();
-
-            if (!in_array($field, $associations)) {
-                return new JsonResponse(
-                    sprintf(
-                        'Unknown association "%s", association does not exist in entity "%s", available associations are "%s".',
-                        $field,
-                        $this->admin->getClass(),
-                        implode(', ', $associations)),
-                    404);
+            $manager = $this->pool->getContainer()->get($admin->getManagerType())->getManager();
+            $classMetadata = $manager->getClassMetadata($admin->getClass());
+
+            if (!$classMetadata->hasAssociation($field)) {
+                return new JsonResponse(sprintf(
+                    'Unknown association "%s", association does not exist in entity "%s", available associations are "%s".',
```

Comment:
Are you sure this is not over 120 chars?

Reply:
well spotted, it's 124 chars long

Reply:
Fixed